frontend/www/js/omegaup/components/course/ViewProgress.test.ts

```diff
@@ -18,44 +18,104 @@ describe('ViewProgress.vue', () => {
-          {
-            alias: 'assignment',
-            assignment_type: 'homework',
-            description: 'Assignment',
-            start_time: new Date(0),
-            finish_time: new Date(),
-            name: 'Assignment',
-            order: 1,
-            scoreboard_url: '',
-            scoreboard_url_admin: '',
-          } as omegaup.Assignment,
-        ],
-        students: <types.StudentProgress[]>[
-          {
-            name: 'student',
-            points: {
-              ['assignment']: { ['problem1']: 100, ['problem2']: 100 },
-            },
-            progress: {
-              ['assignment']: { ['problem1']: 55, ['problem2']: 44 },
-            },
-            score: {
-              ['assignment']: { ['problem1']: 55, ['problem2']: 44 },
-            },
-            username: 'student',
-          } as types.StudentProgress,
-        ],
-      },
+      propsData: baseViewProgressProps,
     });
     expect(wrapper.text()).toContain(
       T.courseStudentsProgressExportToSpreadsheet,
     );
   });
+
+  it('Should handle escaped csv cells', () => {
+    const wrapper = shallowMount(course_ViewProgress, {
+      propsData: baseViewProgressProps,
+    });
+    const escapedCell = escapeCsv('Escaped "text"');
+    expect(escapedCell).toBe('"Escaped ""text""');
+  });
+
+  it('Should handle escaped xml cells', () => {
+    const wrapper = shallowMount(course_ViewProgress, {
+      propsData: baseViewProgressProps,
+    });
+    const escapedXml = escapeXml('Escaped <text>');
+    expect(escapedXml).toBe('Escaped &lt;text&gt;');
+  });
+
+  it('Should handle ods content', () => {
+    const wrapper = shallowMount(course_ViewProgress, {
+      propsData: baseViewProgressProps,
+    });
+
+    const odsContent = toOds(courseName, wrapper.vm.progressTable);
+    expect(odsContent).toBe(`<table:table table:name="${courseName}">
+<table:table-column table:number-columns-repeated="3"/>
+<table:table-row>
+<table:table-cell office:value-type="string"><text:p>${T.profileUsername}\
+</text:p></table:table-cell><table:table-cell office:value-type="string">\
+<text:p>${T.wordsName}</text:p></table:table-cell><table:table-cell \
+office:value-type="string"><text:p>${assignment.name}</text:p>\
+</table:table-cell></table:table-row>
+<table:table-row>
+<table:table-cell office:value-type="string"><text:p>${student.username}\
+</text:p></table:table-cell><table:table-cell office:value-type="string">\
+<text:p>${student.name}</text:p></table:table-cell><table:table-cell \
+office:value-type="float" office:value="${score}"><text:p>${score}</text:p>\
+</table:table-cell></table:table-row>
+</table:table>`);
+  });
+
+  it('Should handle csv contents', () => {
+    const wrapper = shallowMount(course_ViewProgress, {
+      propsData: baseViewProgressProps,
+    });
+
+    const csvContent = toCsv(wrapper.vm.progressTable);
+    expect(csvContent)
+      .toBe(`${T.profileUsername},${T.wordsName},${assignment.name}
```

Comment:
```suggestion
      .toBe(`${T.profileUsername},${T.wordsName},${assignment.name}\r
```

si te fijas en https://github.com/omegaup/omegaup/blob/33b316fcb92bfec62dd3f3c071ac71be951dcd13/frontend/www/js/omegaup/components/course/ViewProgress.vue#L120, las filas se están uniendo con `\r\n` (fin de línea de Windows), no `\n`(fin de línea de todos los demás).

Reply:
Gracias.